util/error_document.go

```diff
@@ -20,3 +27,19 @@ func ErrorDoc(message, status string) *ErrorDocument {
 		}},
 	}
 }
+
+type ErrNotFound struct {
+	Type string
+}
+
+func (e ErrNotFound) Error() string {
+	return fmt.Sprintf("%s not found", e.Type)
+}
+
+func (e ErrNotFound) Is(err error) bool {
+	return reflect.TypeOf(err) == reflect.TypeOf(e)
+}
```

Comment:
I don't fully understand the purpose of this function. The thing I understand is that I can check if another error is of the type "ErrNotFound". But wouldn't be using `errors.Is(someError, util.ErrNotFoundEmpty)` instead?

Reply:
Oh, that's very interesting! Thank you!